src/Sarif.Converters/ContrastSecurityConverter.cs

```diff
@@ -324,7 +325,46 @@ private Result ConstructAuthorizationRulesMissingDenyResult(ContrastLogReader.Co
 
         private Result ConstructInsecureHashAlgorithmsResult(ContrastLogReader.Context context)
         {
-            return CreateResultCore(context);
+            Result result = CreateResultCore(context);
+
+            Stack stack = context.MethodEvent.Stack;
+            result.Stacks = new List<Stack>
+            {
+                stack
+            };
+
+            Location resultLocation = stack.Frames[1].Location;
+            result.Locations = new List<Location>
+            {
+                resultLocation
+            };
+
+            string insecureClassName = GetClassNameFromCtorCall(
+                stack.Frames[0].Location.LogicalLocation.FullyQualifiedName);
+            result.Message = new Message
+            {
+                Id = "default",
+                Arguments = new List<string>
+                {
+                    resultLocation.LogicalLocation.FullyQualifiedName,
+                    insecureClassName
+                }
+            };
+
+            return result;
+        }
+
+        private static readonly Regex ClassNameFromCtorRegex = new Regex(@"\.([^.]+)\.\.ctor\(\)$");
```

Comment:
>ClassNameFromCtorRegex [](start = 38, length = 22)

you should make this a compiled regex with explicit capture for best performance.

Reply:
Fixed.

---
In reply to: [281669231](https://github.com/Microsoft/sarif-sdk/pull/1477#discussion_r281669231) [](ancestors = 281669231)